index/indexer.go

```diff
@@ -464,6 +465,26 @@ func (e *indexer) indexUses(p *packages.Package, fi *fileInfo, filename string)
 				return fmt.Errorf(`emit moniker": %v`, err)
 			}
 
+			// Emit a reference result edge and create a small set of edges that link
+			// the reference result to the range (and vice versa). This is necessary to
+			// mark this range as a reference to _something_, even though the definition
+			// does not exist in this source code.
+
+			refResultID, err := e.emitReferenceResult()
+			if err != nil {
+				return fmt.Errorf(`emit "referenceResult": %v`, err)
+			}
+
+			_, err = e.emitTextDocumentReferences(rangeID, refResultID)
```

Comment:
Does the spec say

`range->referenceResult`

is a valid relationship? I thought it had to go through a resultSet:

`range->resultSet->referenceResult`

Are you thinking of monikers? I recall us chatting about that yesterday. Wait a second, is

`range->moniker`

even a valid relationship? Huh, the spec says in prose it's valid:

> ...LSIF introduces optional monikers which can be linked to ranges using a corresponding edge.
> ...with a moniker (e.g. a handle in string format) that is bound to the corresponding range declaration.

But all of the examples use a `resultSet` between the range and moniker. 😕